src/printing.jl

```diff
@@ -192,5 +192,468 @@ function show(stream::IO, x::Union{LazySum, LazyProduct})
+    end
+    indr, indc = indices(X,1), indices(X,2)
+    nr, nc = length(indr), length(indc)
+    rdots, cdots = false, false
+    rr1, rr2 = UnitRange{Int}(indr), 1:0
+    cr1, cr2 = UnitRange{Int}(indc), 1:0
+    if limit
+        if nr > 4
+            rr1, rr2 = rr1[1:2], rr1[nr-1:nr]
+            rdots = true
+        end
+        if nc > 4
+            cr1, cr2 = cr1[1:2], cr1[nc-1:nc]
+            cdots = true
+        end
+    end
+    print(io, prefix, "[")
+    for rr in (rr1, rr2)
+        for i in rr
+            for cr in (cr1, cr2)
+                for j in cr
+                    j > first(cr) && print(io, " ")
+                    if !isassigned(X,Int(mirror_world_index(i, ldims)),Int(mirror_world_index(j, rdims)))
+                        print(io, undef_ref_str)
+                    else
+                        el = X[mirror_world_index(i, ldims), mirror_world_index(j, rdims)]
+                        show(io, el)
+                    end
+                end
+                if last(cr) == last(indc)
+                    i < last(indr) && print(io, "; ")
+                elseif cdots
+                    print(io, " \u2026 ")
+                end
+            end
+        end
+        last(rr) != nr && rdots && print(io, "\u2026 ; ")
+    end
+    print(io, "]")
+end
+
+
+function showarray_stdord(io::IO, X::AbstractVecOrMat, ldims::Vector, rdims::Vector, repr::Bool = true; header = true)
+    if repr && ndims(X) == 1
+        return show_vector_std(io, X, ldims, "[", "]")
+    end
+    if !haskey(io, :compact)
+        io = IOContext(io, :compact => true)
+    end
+    if !repr && get(io, :limit, false) && eltype(X) === Method
+        # override usual show method for Vector{Method}: don't abbreviate long lists
+        io = IOContext(io, :limit => false)
+    end
+    (!repr && header) && print(io, summary(X))
+    if !isempty(X)
+        (!repr && header) && println(io, ":")
+        if ndims(X) == 0
+            if isassigned(X)
+                return show(io, X[])
+            else
+                return print(io, undef_ref_str)
+            end
+        end
+        if repr
+            print_matrix_repr_std(io, X, ldims, rdims)
+        else
+            punct = (" ", "  ", "")
+            print_matrix_std(io, X, ldims, rdims, punct...)
+        end
+    elseif repr
+        Base.repremptyarray(io, X)
+    end
+end
+showarray_stdord(io::IO, X::Vector, dims::Vector, repr::Bool = true; header = true) = showarray_stdord(io, X, dims, [1], repr; header = header)
+
+
+function showsparsearray_stdord(io::IO, S::SparseMatrixCSC, ldims::Vector, rdims::Vector)
+    if nnz(S) == 0
+        return Base.show(io, MIME("text/plain"), S)
+    end
+
+    limit::Bool = get(io, :limit, false)
+    if limit
+        rows = displaysize(io)[1]
+        half_screen_rows = div(rows - 8, 2)
+    else
+        half_screen_rows = typemax(Int)
+    end
+    pad = ndigits(max(S.m,S.n))
+    k = 0
```

Comment:
The assignment of `k` here is overwritten later when `k` is used as loop variable, so this line here is not needed.